nvflare/fuel/utils/dict_utils.py

```diff
@@ -120,3 +120,70 @@ def augment(to_dict: dict, from_dict: dict, from_override_to=False) -> str:
             to_dict[k] = fv
 
     return ""
+
+
+def _update_component_dict(comp_list: list, target: dict) -> str:
+    for c in comp_list:
+        if not isinstance(c, dict):
+            return f"component must be a dict but got {type(c)}"
```

Comment:
should we change these checks and all checks throughout the code base to use those "validation_utils"?